React/creating-components.md

```diff
@@ -0,0 +1,79 @@
+# Creating Components
+
+## Learning Goals
+- Create our own component
+- Link and render a component from an existing component
+
+Now that we've gotten started with a basic React application, we want to be able to extend it with our own components. We create new components within our applications to encapsulate logic and HTML related to a specific concept or portion of our site. Note that up until this point, we have had very clear lines between HTML, CSS and JavaScript because they're usually in separate files. With React components, these lines can get a little bit blurry since we will potentially have all of these things in a single component file.
+
+Each component will be a small, reusable piece of code that manages their own **content, presentation and behavior**. 
+
+### Overview
+It is important to understand what is happening currently within the React application so we can model the same pattern with a custom component. Right now, our React application renders the `App` component from the `index.js` file and puts it in the DOM element where the ID is "root". Normally, this is only done **once** in a React application, so we are going to leave this code alone as we build on.
+
+Next, our `App` component contains a render function will which render a `header` and a `p` using JSX.
+
+The powerful thing about React is that **any component can render any other component**. That is, we can (and should) **nest our components** within one another to create the most effective React application design.
```

Comment:
We touch upon the idea of nesting components in a few spots in this textbook resource (including talking about it as the key takeaway), however I don't feel like it's really getting sufficient emphasis. Especially since the next activity asks them to break down a website into a hierarchy of nested components.

We should at the very least make it clear that nesting components isn't just a powerful aspect of React's design, but is a necessary structure for modelling a web page. Because web pages are about representing a physical space (a flat 2D plane, like a piece of paper), it is possible to consider every location/element within that space as being contained in one or more larger spaces -- much like a map of the US has states, and within them has counties, and cities, and streets, etc.

We could also make the comparison to HTML itself, which forces a hierarchical/nested structure in the language itself. Every element on the page in a React app is the result of a single component's `render()` function, and because elements in HTML are necessarily contained inside of other elements, our components must be nested in a way that reflects the same structure as the HTML.